sdks/java/io/cdap/src/test/java/org/apache/beam/sdk/io/cdap/CdapIOTest.java

```diff
@@ -154,11 +171,45 @@ public void testReadingData() {
     for (int i = 1; i < EmployeeInputFormat.NUM_OF_TEST_EMPLOYEE_RECORDS; i++) {
       expected.add(KV.of(String.valueOf(i), EmployeeInputFormat.EMPLOYEE_NAME_PREFIX + i));
     }
-    PCollection<KV<String, String>> actual = p.apply("ReadTest", read);
+    PCollection<KV<String, String>> actual = p.apply("ReadBatchTest", read);
     PAssert.that(actual).containsInAnyOrder(expected);
     p.run();
   }
 
+  @Test
+  public void testReadingDataStreaming() {
+    DirectOptions options = PipelineOptionsFactory.as(DirectOptions.class);
+    options.setBlockOnRun(false);
+    options.setRunner(DirectRunner.class);
+    Pipeline p = Pipeline.create(options);
+
+    EmployeeConfig pluginConfig =
+        new ConfigWrapper<>(EmployeeConfig.class).withParams(TEST_EMPLOYEE_PARAMS_MAP).build();
+    MappingUtils.registerStreamingPlugin(
+        EmployeeStreamingSource.class,
+        Long::valueOf,
+        new ReceiverBuilder<>(EmployeeReceiver.class).withConstructorArgs(pluginConfig));
+
+    CdapIO.Read<String, String> read =
+        CdapIO.<String, String>read()
+            .withCdapPlugin(Plugin.createStreaming(EmployeeStreamingSource.class))
+            .withPluginConfig(pluginConfig)
+            .withKeyClass(String.class)
+            .withValueClass(String.class);
+
+    List<String> storedRecords = EmployeeReceiver.getStoredRecords();
+    List<String> outputRecords = TestOutputDoFn.getRecords();
+    outputRecords.clear();
+
+    p.apply("ReadStreamingTest", read)
+        .setCoder(KvCoder.of(NullableCoder.of(StringUtf8Coder.of()), StringUtf8Coder.of()))
+        .apply(Values.create())
+        .apply(ParDo.of(new TestOutputDoFn()));
+
+    p.run().waitUntilFinish(Duration.standardSeconds(15));
+    assertEquals(storedRecords, outputRecords);
```

Comment:
Instead of comparing with a local assert, we should update the pipeline to use PAssert. (otherwise, this test will not work for other runners).

https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java

Reply:
Changed to PAssert